src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -134,7 +134,11 @@ private MaxwellAbstractRowsEvent processRowsEvent(AbstractRowEvent e) throws Sch
 
 		table = tableCache.getTable(e.getTableId());
 
-		if (table == null) {
+		if ( table == null && this.filter.hasTableBlacklist() ) {
+			LOGGER.warn(String.format("couldn't find table in cache for table id: %d, assuming blacklisted table", e.getTableId()));
+			return null;
```

Comment:
ah, no, you misunderstand a bit.  Each transaction is preceded by a TABLE_MAP_EVENT which will contain the database name + table name.